### ISSUE TITLE
Add Async Flavor Interfaces

### ISSUE THREAD

**Issue by aptxkid** [box/box-python-sdk#14]
e.g. client.folder(folder_id='0').upload_async('/path/to/file'), which returns a promise.

**Comment by mattwiller** [comment]
Rather than splitting the SDK interface into both sync and async interfaces, we would probably want to decide to move the entire SDK over to async.  Since this sort of functionality was added at the language level in Python 3.4, this would likely be something we'll tackle when the SDK drops support for Python 2.7 in the next major version.  I'm going to mark this breaking for now so we don't lose track of the idea, but it's unlikely to land on the v2.x branch.

**Comment by sinwoobang** [comment]
Any progress on this?

**Comment by sujaygarlanka** [comment]
@sinwoobang Our team is currently short staffed and there is no plan to do this anytime soon. However, we are growing our team over the next few months and may revisit this then. Sorry for any inconvenience!

**Comment by oneryalcin** [comment]
@sujaygarlanka  yes, I'd like to see async support on your official python API as it limits the usability (doesn't want to go multi threads in python)
